File: java/oop/CollegeApp/src/com/techlabs/college/Professor.java
package com.techlabs.college;

import java.text.ParseException;

public class Professor extends Person {
	private Branch branch;
	private int lectures;
	private double salary;
	private int semester;

	public Professor(int id, String name, String address, String dob,Branch branch, int lectures, double salary, int semester) throws ParseException {
		super(id, name, address, dob);
		this.branch=branch;
		this.lectures=lectures;
		this.salary=salary;
		this.semester=semester;
	}

	public String getBranch() {
		return this.branch.name();
	}

	public double getSalary() {
		return this.salary;
	}

	public int getSemester() {
		return this.semester;
	}

	public String giveSalaryLogic() {
		return "No. of lectures taken: " + this.lectures + "\nTotal salary: "
				+ calculateSalary();
	}

	private double calculateSalary() {
		return this.salary * this.lectures;
	}
}

File: dotNet/CSharp/CollegeApp/CollegeApp/Professor.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CollegeApp
{
    class Professor : Person, SalariedEmployee
    {
        private Branch _branch;
        private int _lectures;
        private double _salary;
        private int _semester;

        public Professor(int id, string name, string address, string dateOfBirth, Branch branch, int lectures,double salary, int semester) : base(id, name, address, dateOfBirth)
        {
            _branch = branch;
            _lectures = lectures;
            _salary = salary;
            _semester = semester;
        }

        public String GetBranch
        {
            get
            {
                return _branch.ToString();
            }
        }

        public double Salary
        {
            get
            {
                return _salary;
            }
        }

        public Int32 Semester
        {
            get
            {
                return _semester;
            }
        }

        public String GiveSalaryLogic()
        {
            return "No. of Lectures taken: " + _lectures
                    + "\n Total salary: "+CalculateSalary();
        }

        private Double CalculateSalary()
        {
            return Salary * _lectures;
        }

        public override string ToString()
        {
            return "Id: " + Id
                +"\n Name: " + Name
                +"\n Address: "+Address
                +"\n DateOfBirth: "+DateOfBirth
                +"\n Branch: "+GetBranch
                +"\n Semester: "+Semester
                +"\n Basic Salary: "+Salary
                +"\n Salary Logic: \n"+GiveSalaryLogic();
        }
    }
}
